scripts/generator_utils.py

```diff
@@ -0,0 +1,211 @@
+import yaml
+from argparse import ArgumentParser
+
+
+def common_args() -> ArgumentParser:
+    """
+    Common argument parser for each script
+    """
+    parser = ArgumentParser()
+    parser.add_argument("ConfigFilePath", metavar='config_file_path', type=str, help="Configuration data file path.")
+    parser.add_argument("--settings", type=str, required=False, help="Settings module")
+    return parser
+
+
+def update_model(model, **kwargs):
+    """
+    Helper function for updaing a django model
+    Args:
+        model: Django Model
+        kwargs: Keyword args for each field values
+    """
+    for attr, val in kwargs.items():
+        setattr(model, attr, val)
+    model.save()
+
+
+def deep_merge(source, destination):
+    """
+    helper function to merge two dictionaries
+    """
+    for key, value in source.items():
+        if isinstance(value, dict):
+            # get node or create one
+            node = destination.setdefault(key, {})
+            deep_merge(value, node)
+        else:
+            destination[key] = value
+
+    return destination
+
+
+class Config:
+    """
+    Configuration generation helper class
+    """
+
+    # key value map of microsites
+    microsites = {}
+
+    # key value map of organiztions
+    organizations = {}
+
+    # main domain to work with
+    main_domain = None
+
+    # holder for storing global override values
+    global_overrides = {
+        'overrides': {},
+        'context_overrides': {},
+    }
+
+    def __init__(self, config):
+        """
+        Initialize Config class
+        """
+        self._config = config
+        self.organizations = config['organizations']
+        self.main_domain = config['main_domain']
+
+        # if there will be a site for each organization
+        if config.get('site_for_each_organization', False):
+            for key, val in config['organizations'].items():
+                self.microsites[key] = {
+                    'name': val['name'],
+                    'overrides': {
+                        'lms': {
+                            'openedx.core.djangoapps.site_configuration.models.SiteConfiguration': {
+                                'site_values': {
+                                    'course_org_filter': key
+                                }
+                            }
+                        }
+                    },
+                    'context_overrides': {},
+                }
+        else:
+            # otherwise microsites can be given seperately from organizations
+            for key, val in config['microsites'].items():
+
+                # $ is a special key and used for global overrides
+                if key == '$':
+                    continue
+
+                self.microsites[key] = {
+                    'name': val['name'],
+                    'overrides': {},
+                    'context_overrides': {},
+                }
```

Comment:
can we break this down into functions?

Reply:
Yes, moved into two separate private methods.